libs/pageserver_api/src/models/filename.rs

```diff
@@ -242,12 +223,8 @@ impl serde::Serialize for LayerFileName {
         S: serde::Serializer,
     {
         match self {
-            LayerFileName::Image(fname) => serializer.serialize_str(&format!("{}", fname)),
-            LayerFileName::Delta(fname) => serializer.serialize_str(&format!("{}", fname)),
-            #[cfg(test)]
-            LayerFileName::Test(t) => {
-                serializer.serialize_str(&format!("{LAYER_FILE_NAME_TEST_PREFIX}{t}"))
-            }
+            LayerFileName::Image(fname) => serializer.serialize_str(&format!("{fname}")),
+            LayerFileName::Delta(fname) => serializer.serialize_str(&format!("{fname}")),
```

Comment:
```suggestion
            LayerFileName::Image(fname) | LayerFileName::Delta(fname) => serializer.collect_str(&fname),
```

But I guess the scope of this PR is already quite much.

Reply:
I don't think they could be united that easily, since one is `DeltaFileName` and another is `ImageFileName`.
So, I'll leave it as is, thank you.

Reply:
Ah yeah didn't realize they were different types as well :) But yeah the point was `collect_str` which collects any Display and doesn't require to_string here.